src/spatial/traversal/IterationDiffAnalyzer.scala

```diff
@@ -0,0 +1,141 @@
+package spatial.traversal
+
+import argon._
+import spatial.lang._
+import spatial.node._
+import spatial.util.spatialConfig
+import spatial.util.modeling._
+import spatial.metadata.control._
+import spatial.metadata.access._
+import spatial.metadata.memory._
+import spatial.metadata.bounds.Expect
+
+case class IterationDiffAnalyzer(IR: State) extends AccelTraversal {
+
+
+  private def visitInnerControl(lhs: Sym[_], rhs: Op[_]): Unit = {
+    dbgs(stm(lhs))
+    val blks = rhs.blocks.map{block => latencyAndInterval(block) }
+    val latency = blks.map(_._1).sum
+    val interval = (1.0 +: blks.map(_._2)).max
+    dbgs(s" - Latency:  $latency")
+    dbgs(s" - Interval: $interval")
+    lhs.bodyLatency = latency
+    lhs.II = lhs.userII.getOrElse(interval)
+  }
+
+  private def findCycles(lhs: Sym[_], ctrl: Control[_]): Unit = {
+    ctrl.bodies.foreach{body =>
+      body.blocks.foreach{case (iters, block) =>
+        val stms = block.stms
+        val cycles = findAccumCycles(stms.toSet).accums
+        if (cycles.nonEmpty) {
+          dbgs(s"\n\nFound cycles in $lhs ($iters): ")
+          cycles.foreach{c => dbgs(s"  $c")}
+          cycles.collect{case AccumTriple(mem,reader,writer) if (mem.isLocalMem && reader != writer) => 
+
+            val read = reader.affineMatrices.head.matrix
+            val write = writer.affineMatrices.head.matrix
+            val diff = write - read
+            if (iters.nonEmpty) {
+              val stride = iters.last.ctrStep match {case Expect(c) => Some(c.toInt); case _ => None}
+              val par = iters.last.ctrParOr1
+              if (stride.isDefined) {
+                // iterDiff between iters
+                val thisIterReads  = List.tabulate(par){i => read.incrementConst(i*stride.get)}  // TODO: Replace this with access.affineMatrices.map(_.matrix)?
+                val thisIterWrites = List.tabulate(par){i => write.incrementConst(i*stride.get)}  // TODO: Replace this with access.affineMatrices.map(_.matrix)?
+                val nextIterReads  = List.tabulate(par){i => read.increment(iters.last,1).incrementConst(i*stride.get)}
+                val diff = thisIterWrites.last - thisIterReads.head // How far is the last write from the first read?
+                val advancePerInc = nextIterReads.head - thisIterReads.head // How far do we advance in one tick?
+                val minIterDiff = diff.collapse.zip(advancePerInc.collapse).map{case (a,b) => a / b}.min
+                dbgs(s"Iteration Diff = ${minIterDiff}")
+                reader.iterDiff = minIterDiff
+                writer.iterDiff = minIterDiff
+                mem.iterDiff = minIterDiff
+                if (par > 1) { 
+                  // iterDiff within iter
+                  /* 
```

Comment:
This is where things get really messy.  I think the idea behind the analysis is right, given the pictures, but no idea how to actually tie this metadata in to unrolling (i.e only vectorize lanes with the same laneWait together in the same access node)